### ISSUE TITLE
History Tab is broken on first load

### ISSUE THREAD

**Issue by eehondas** [dnadesign/silverstripe-elemental#390]
Hello,
I was looking around at the improvements that have been shipped with Silverstripe 4.2.
First of all thank you for the implementation of the History Tab on the Elemental Page and on the elements itself.
I think the functionality is very good from a user standpoint.
I believe there is a small bug on the History Tab of an element detailed view (say ElementContent for example):
The history tab is accessible before the element is saved for the first time (created), because of this when trying to access the History Tab on first load it the CMS gets stuck in an infinite loop of loading (I assume because it is looking for Versions of that specific object that don't exist yet).
Anyway, I believe the fix is as quick as checking for the presence of an ID here:
```php
            // Support for new history viewer in SS 4.2+
            if (class_exists(HistoryViewerField::class)) {
                Requirements::javascript('dnadesign/silverstripe-elemental:client/dist/js/bundle.js');

                $historyViewer = HistoryViewerField::create('ElementHistory');
                $fields->addFieldToTab('Root.History', $historyViewer);

                $fields->fieldByName('Root.History')
                    ->addExtraClass('elemental-block__history-tab tab--history-viewer');
            }
```
While I'm at it I wanted to ask: seeing that the History functionality of a normal Silverstripe Page and an Elemental Page are now completely different, do you guys plan to roll-out this more-advanced functionality with the preview to the SiteTree History Tab itself? Are there plans for the HistoryViewerField to be used in a normal DataObject?

Thanks guys.

**Comment by robbieaverill** [comment]
Hi @eehondas, thank you for the feedback!

Yes you’re right, we should check the element exists before pushing the history field in getCMSFields. Would you mind making a PR for it?

With SilverStripe 4.3 the new history viewer will be enabled by default on all pages. We didn’t enable it by default in 4.2 because it doesn’t have compare mode at that version, so is a feature regression over the legacy history controller. You can still opt into it everywhere if you want to, by enabling an extension (see the example in this repository)

**Comment by ScopeyNZ** [comment]
Hi @eehondas. 3.0 is perfect. We can release this fix as a bugfix (eg. `3.0.1`) and we can merge up to `3` for another minor release if applicable (`3.1.0`). Master is for the next major (`4.0.0`). Hopefully that makes the branching strategy a little clearer?

**Comment by ScopeyNZ** [comment]
Closed in #391
